polyply/src/nonbond_engine.py

```diff
@@ -125,6 +125,12 @@ def copy(self):
                                  boxsize=self.boxsize)
         return new_obj
 
+    def concatenate_trees(self):
+        self.defined_idxs = [list(np.where(self.positions[:, 0].reshape(-1) != np.inf)[0])]
```

Comment:
Under the assumption that if the x coordinate is inf, so is the rest (and vice versa)?
Consider using np.any with the appropriate axis instead, but also fine to leave it as is.
I also do not understand the `list` call, together with the indexing below

Reply:
defined indices is a list of lists, where each list contains the indices stored in the tree list of the same index (i.e. the first list in self.defined_idxs contains all coordinate idxs that are stored in the first tree etc.). Here we only have a single tree so we overwrite the list of lists into a new list of lists with one element. The idxs[0] is due to no.where returning a tuple. I'm sure we can condense some of the bookkeeping but I think this PR has been in the making long enough.